test/th5_system.c

```diff
@@ -447,21 +447,58 @@ test_h5_strcasestr(void)
 
     /* check that H5_strcasestr returns target in empty search */
     str = H5_strcasestr(haystack, "");
-    CHECK_PTR_EQ(str, haystack, "H5_strcasestr search for empty");
+    CHECK_PTR(str, "H5_strcasestr for empty substring");
+    if (str)
+        VERIFY_STR(str, haystack, "comparing H5_strcasestr to original string for empty substring");
 
     /* Check that H5_strcasestr find a string of same case */
     str = H5_strcasestr(haystack, "string");
-    CHECK_PTR_EQ(str, &(haystack[8]), "H5_strcasestr search same case");
+    CHECK_PTR(str, "H5_strcasestr for substring of same case");
+    if (str)
+        VERIFY_STR(str, "string", "comparing H5_strcasestr for substring of same case");
 
     /* Check that H5_strcasestr find a string of different case */
     str = H5_strcasestr(haystack, "sTrInG");
-    CHECK_PTR_EQ(str, &(haystack[8]), "H5_strcasestr search different case");
+    CHECK_PTR(str, "H5_strcasestr for substring of different case");
+    if (str)
+        VERIFY_STR(str, "string", "comparing H5_strcasestr for substring of different case");
```

Comment:
I think the 2nd parameter should be "sTrInG"?

Reply:
In `VERIFY_STR` we're checking to make sure that `H5_strcasestr` returns the actual original string of "string" from `haystack`. We just pass "sTrInG" to `H5_strcasestr` to make sure that it can find that portion even though the search uses a string with weird casing to try and find it.